executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java

```diff
@@ -0,0 +1,65 @@
+package executor.service.facade.execution;
+
+import com.google.common.base.Supplier;
+import executor.service.holder.ScenarioQueueHolder;
+import executor.service.model.Scenario;
+import executor.service.model.Step;
+import executor.service.service.executor.ScenarioExecutor;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openqa.selenium.WebDriver;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import static org.mockito.Mockito.*;
+
+class ExecutionServiceImplTest {
+    @Mock
+    private ScenarioExecutor scenarioExecutor;
+
+    ScenarioQueueHolder scenarioQueueHolder;
+    private ExecutionService executionService;
+    private BlockingQueue<Scenario> scenarioQueue;
+
+    private AutoCloseable closeable;
+
+    @BeforeEach
+    void setUp() throws IllegalAccessException {
+        closeable = MockitoAnnotations.openMocks(this);
+        scenarioQueueHolder = new ScenarioQueueHolder();
+        scenarioQueue = scenarioQueueHolder.getQueue();
+        executionService = new ExecutionServiceImpl(scenarioQueueHolder, scenarioExecutor);
+    }
+
+    @AfterEach
+    void cleanUp() throws Exception {
+        closeable.close();
+    }
+    @Test
+    @DisplayName("Test execute method with interruptions")
+    void execute() throws InterruptedException {
+        WebDriver mockWebDriver = mock(WebDriver.class);
+        Supplier<WebDriver> webDriverSupplier = () -> mockWebDriver;
```

Comment:
Guava's functional primitives can be replaced with Java API
`import java.util.function.Supplier;`

Reply:
Thanks, done